src/Control/CookieJar.php

```diff
@@ -63,26 +64,31 @@ public function __construct($cookies = array())
      *
      * @param string $name The name of the cookie
      * @param string $value The value for the cookie to hold
-     * @param int $expiry The number of days until expiry; 0 indicates a cookie valid for the current session
+     * @param int $expiry A Unix timestamp of the time the cookie expires; 0 expires at the end of the current session
      * @param string $path The path to save the cookie on (falls back to site base)
      * @param string $domain The domain to make the cookie available on
      * @param boolean $secure Can the cookie only be sent over SSL?
      * @param boolean $httpOnly Prevent the cookie being accessible by JS
      */
-    public function set($name, $value, $expiry = 90, $path = null, $domain = null, $secure = false, $httpOnly = true)
+    public function set($name, $value, $expiry = 0, $path = null, $domain = null, $secure = false, $httpOnly = true)
     {
+        if ($expiry > 0 && $expiry < time()) {
+            Deprecation::notice(
+                '5.0',
+                'Cookie::set() requires $expiry to be a Unix timestamp of the time the cookie expires'
+            );
```

Comment:
```suggestion
            );
            $expiry = DBDatetime::now()->getTimestamp() + (86400 * $expiry);
```

Reply:
I was wondering about this. Having a closer look at it, I'm assuming it's to hook into the `DBDatetime::$mock_now`?